ENIGMAsystem/SHELL/Graphics_Systems/OpenGLES/wrappers/shader_wrappers.cpp

```diff
@@ -0,0 +1,50 @@
+#include "shader_wrappers.h"
+
+void glUniform1ui(int location, unsigned v0)
+{
+  glUniform1i(location ,v0);
+  
+}
+
+void glUniform2ui(int location, unsigned v0, unsigned v1)
+{
+  glUniform2i(location , v0,v1);
+  
+}
+
+void glUniform3ui(int location, unsigned v0, unsigned v1, unsigned v2)
+{
+  glUniform3i(location,v0,v1,v2);
+  
+}
+
+void glUniform4ui(int location, unsigned v0, unsigned v1, unsigned v2, unsigned v3)
+{
+  glUniform4i(location, v0, v1, v2, v3);
+  
+}
+
+//for unsigned int uniforms 
+
+void glUniform1uiv(int location, int size, const GLuint* value)
+{
+  glUniform1iv(location,size, (const GLint*)value);
+  
+}
+
+void glUniform2uiv(int location, int size, const GLuint* value)
+{
+  glUniform2iv(location,size,(const GLint*)value);
+  
+}
+
+void glUniform3uiv(int location, int size,const GLuint* value)
+{
+  glUniform3iv(location,size,(const GLint*)value);
+  
+}
+
+void glUniform4uiv(int location, int size, const GLuint* value)
+{
+  glUniform4iv(location,size,(const GLint*)value);  
+}
```

Comment:
We're actually going to need a better solution to this one. Passing unsigned as signed ints is just going to overflow. GameMaker Studio 2, being dynamically typed, only exposes `shader_set_uniform_i` and doesn't actually expose the unsigned version of these functions. What we actually need to do here is split it so that only GL3/GLES3 and newer use the unsigned functions. Then we need to completely remove them from GLES2/GL1.

Reply:
I think this is fine. If you're really worried about overflows just put a std::min(v, int_max) or whatever.

Reply:
@RobertBColton @fundies  I think I will look into this after GSoC